src/actions.js

```diff
@@ -1,12 +1,14 @@
 import _ from 'lodash/fp'
 import { encode } from './util/tree'
+import { setState } from './state'
 import { pullOn } from 'futil-js'
 
 export default ({ getNode, flat, dispatch, snapshot, extend }) => ({
   async add(parentPath, value) {
     let target = getNode(parentPath)
     let path = [...parentPath, value.key]
-    extend(value, { path })
+    setState([value], extend)
+    value.path = path
```

Comment:
We should use extend everywhere we do mutations so we can support stuff like knockout observables (or anything that isn’t just and =)